web/src/views/public/tracks/Show.vue

```diff
@@ -0,0 +1,300 @@
+<template>
+  <div>
+    <div class="track-hero" :style="{'background-color': background, color: textColor}">
+      <div class="track-hero__content">
+        <div class="track-hero__left">
+          <div class="track-hero__avatar" :elevation="2">
+            <v-avatar size="120px" class="white" tile>
+              <img :src="album.artwork" :alt="album.name" />
+            </v-avatar>
+          </div>
+          <div class="track-hero__text">
+            <h4 class="track-hero__title">{{ track.title }}</h4>
+            <div class="track-hero__meta">
+              <p>{{ reciter.name }}</p>
+              <p>{{ album.year }} &bull; {{ album.title }}</p>
+            </div>
+          </div>
+        </div>
+        <div class="track-hero__actions">
+          <ul>
+            <li>
+              <v-btn icon class="white--text">
+                <v-icon>add_to_photos</v-icon>
+              </v-btn>
+            </li>
+            <li>
+              <v-btn icon class="white--text">
+                <v-icon>share</v-icon>
+              </v-btn>
+            </li>
+            <li>
+              <v-btn icon class="white--text">
+                <v-icon>print</v-icon>
+              </v-btn>
+            </li>
+            <li>
+              <v-btn icon class="white--text">
+                <v-icon>more_horiz</v-icon>
+              </v-btn>
+            </li>
+          </ul>
+        </div>
+      </div>
+    </div>
+
+    <div class="track-page-content">
+      <v-container grid-list-xl>
+        <v-layout row>
+          <v-flex md7>
+            <v-card class="track-page-content__card track-page-content__card--lyrics lyrics">
+              <div class="lyrics__content" v-if="track.lyrics">
+                <p v-html="track.lyrics.content"></p>
+              </div>
+              <div class="lyrics__empty" v-else>
+                <div class="lyrics__empty-message">We don't have a write-up for this nawha yet.</div>
+              </div>
+            </v-card>
+          </v-flex>
+          <v-flex md5>
+            <v-card class="track-page-content__card track-page-content__card--audio">
+              Audio
+              <section>
+                <p>There is no track available yet</p>
+              </section>
+            </v-card>
+            <v-card class="track-page-content__card track-page-content__card--audio">
+              <section>
+                <p>There is no video available</p>
+              </section>
+            </v-card>
+            <v-card class="track-page-content__card track-page-content__card--album">
+              More
+            </v-card>
+          </v-flex>
+        </v-layout>
+      </v-container>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
```

Comment:
This isn't compiling because we can't use TS with this style of writing components. We'll need to remove `lang="ts"` for now while we refactor these components later.

Reply:
Specifically, we're getting this sort of errors: 

```
ERROR in /Users/zainmehdi/Code/nawhas/nawhas.com/web/src/views/public/tracks/Show.vue(127,10):
127:10 Property 'setBackgroundFromImage' does not exist on type '{ name: string; data(): { background: string; textColor: string; }; computed: { reciter: Computed; album: Computed; track: Computed; }; methods: { setBackgroundFromImage(): void; }; created(): void; beforeRouteEnter(to: any, from: any, next: any): Promise<...>; beforeRouteUpdate(to: any, from: any, next: any): Promi...'.
    125 |   },
    126 |   created() {
  > 127 |     this.setBackgroundFromImage();
        |          ^
    128 |   },
    129 |   async beforeRouteEnter(to, from, next) {
    130 |     await fetchData(to.params.reciter, to.params.album, to.params.track);

ERROR in /Users/zainmehdi/Code/nawhas/nawhas.com/web/src/views/public/tracks/Show.vue(122,16):
122:16 Property 'textColor' does not exist on type '{ setBackgroundFromImage(): void; }'.
    120 |           }
    121 |           this.background = swatch.getHex();
  > 122 |           this.textColor = swatch.getBodyTextColor();
        |                ^
    123 |         });
    124 |     },
    125 |   },

 error  in /Users/zainmehdi/Code/nawhas/nawhas.com/web/src/views/public/tracks/Show.vue
```